test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/CompleteTagHelperDescriptorComparer.cs

```diff
@@ -23,16 +23,25 @@ bool IEqualityComparer<TagHelperDescriptor>.Equals(TagHelperDescriptor descripto
             return base.Equals(descriptorX, descriptorY) &&
                    // Tests should be exact casing
                    string.Equals(descriptorX.TagName, descriptorY.TagName, StringComparison.Ordinal) &&
-                   descriptorX.Attributes.SequenceEqual(descriptorY.Attributes,
-                                                        CompleteTagHelperAttributeDescriptorComparer.Default);
+                   string.Equals(descriptorX.Prefix, descriptorY.Prefix, StringComparison.Ordinal) &&
+                   Enumerable.SequenceEqual(
+                       descriptorX.RequiredAttributes,
+                       descriptorY.RequiredAttributes,
+                       StringComparer.OrdinalIgnoreCase) &&
+                   descriptorX.Attributes.SequenceEqual(
+                       descriptorY.Attributes,
+                        CompleteTagHelperAttributeDescriptorComparer.Default);
         }
 
         int IEqualityComparer<TagHelperDescriptor>.GetHashCode(TagHelperDescriptor descriptor)
         {
-            return HashCodeCombiner.Start()
-                                   .Add(base.GetHashCode())
-                                   .Add(descriptor.Attributes)
-                                   .CombinedHash;
+            return HashCodeCombiner
+                .Start()
+                .Add(base.GetHashCode())
```

Comment:
:-1: it's a test class, don't want to maintain this this thing everytime the real implementation changes.